org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -2244,6 +2251,10 @@ protected void setUpGeneralParameters() {
             // So that each separate compile knows about modal reactors, do this:
             targetConfig.compileDefinitions.put("MODAL_REACTORS", "");
         }
+        if(targetConfig.threading && targetConfig.platform == Platform.ARDUINO) {
+            targetConfig.threading = false;
+            System.out.println("******** Threading is incompatible on Arduino...Turning off LF threading support.");
```

Comment:
We have an `errorReporter` that you could use to report warnings (although I think this should be an error instead of overriding the `threading` target property).

Reply:
The idea here was to not have to write threading=false inside the lf file since no arduino program will ever have threading.

Reply:
If the behavior was the opposite (threading is false by default, true if enabled), then I think it would make more sense to error.

Reply:
There is a `targetConfig.setByUser` that can help here. If threading is set by user, then I think an error is more appropriate.